### ISSUE TITLE
Maximum call stack size exceeded with over 139000 records

### ISSUE THREAD

**Issue by 0fork** [orientechnologies/orientjs#116]
I'm running into maximum call stack problems when trying to get large number of paths (139644 to be exact) from the database.

I'm using simple `db.query(q).then(...)` code for querying. This works great with over 80000 records but it seems 139000 is a bit much. I can't chop up the query into smaller pieces and in any case I should be able to get this amount of records from the database. Anybody know how to deal with this?

```
/Dev/Main/node_modules/orientjs/lib/transport/binary/protocol28/operation.js:773
        this.readOps.push.apply(this.readOps, remainingOps);
                          ^
RangeError: Maximum call stack size exceeded
    at null.<anonymous> (/Dev/Main/node_modules/orientjs/lib/transport/binary/protocol28/operation.js:773:27)
    at Operation.parseString (/Dev/Main/node_modules/orientjs/lib/transport/binary/protocol28/operation.js:694:12)
    at Operation.consume (/Dev/Main/node_modules/orientjs/lib/transport/binary/protocol28/operation.js:469:35)
    at Connection.process (/Dev/Main/node_modules/orientjs/lib/transport/binary/connection.js:380:17)
    at Connection.handleSocketData (/Dev/Main/node_modules/orientjs/lib/transport/binary/connection.js:281:17)
    at Socket.emit (events.js:95:17)
    at Socket.<anonymous> (_stream_readable.js:765:14)
    at Socket.emit (events.js:92:17)
    at emitReadable_ (_stream_readable.js:427:10)
    at emitReadable (_stream_readable.js:423:5)
    at readableAddChunk (_stream_readable.js:166:9)
    at Socket.Readable.push (_stream_readable.js:128:10)
    at TCP.onread (net.js:529:21)
```

**Comment by maggiolo00** [comment]
@zero- 

can you try to increase the stack size of node?

http://stackoverflow.com/questions/11332422/how-can-i-increase-the-maximum-call-stack-size-in-node-js

Meanwhile i will try to replicate the issue

**Comment by zero-** [comment]
Good tip @maggiolo00, increased stack size to 2000 and I was able to get it done :+1: This is a cron job so I can easily incorporate increasing the stack into it.

**Comment by maggiolo00** [comment]
Cool @zero- 

Can i close it?

**Comment by zero-** [comment]
Not a problem for me anymore so you can close it unless you want to keep it as enhancement etc. At some point it would be nice if the driver would be able to parse large number of records without having to manually increase the stack size. Maybe support for streaming at some point will fix this. But in any case, thanks again :)

**Comment by maggiolo00** [comment]
Ok I will keep as enhancement  :D

**Comment by lvca** [comment]
Can we add this in the docs?

**Comment by maggiolo00** [comment]
Sure,
i will add a section in the docs.

**Comment by hilkeheremans** [comment]
+1; Increasing the stack size is just a workaround; the proper fix would still be to add full streaming support.

**Comment by kurtcorbett** [comment]
+1 for streaming support

**Comment by seeden** [comment]
+1

**Comment by arikalon** [comment]
+1

**Comment by sinfex** [comment]
+1

**Comment by StarpTech** [comment]
+1

**Comment by StarpTech** [comment]
Isn't it possible to create temporary workaround with a custom paging logic?
In style of reactive extension.

https://github.com/Reactive-Extensions/RxJS

**Comment by greenlikeorange** [comment]
Is there any update for this issue ?

**Comment by maggiolo00** [comment]
hi @greenlikeorange 

it's in the roadmap of OrientJS 3.0, that will provide streaming using the new streaming protocol of OrientDB 3.0

**Comment by lvca** [comment]
Implemented in v3.0.0-M2.
